Developer setup instructions on Mac

Running through the dev setup instructions on Mac OS X: following the instructions in the Readme with Ansible version 2.1.1.0 and Vagrant 1.8.5, I run into a bump at `make dev`:

```
$ make dev
make ansible
ansible-galaxy install -r ansible/requirements.yml -p ansible/roles
- downloading role 'nodejs', owned by geerlingguy
- downloading role from https://github.com/geerlingguy/ansible-role-nodejs/archive/4.0.0.tar.gz
- extracting geerlingguy.nodejs to ansible/roles/geerlingguy.nodejs
- geerlingguy.nodejs was installed successfully
vagrant up --provision
Bringing machine 'securethenews' up with 'docker' provider...
==> securethenews: Box 'debian/contrib-jessie64' could not be found. Attempting to find and install...
    securethenews: Box Provider: docker
    securethenews: Box Version: >= 0
==> securethenews: Loading metadata for box 'debian/contrib-jessie64'
    securethenews: URL: https://atlas.hashicorp.com/debian/contrib-jessie64
The box you're attempting to add doesn't support the provider
you requested. Please find an alternate box or use an alternate
provider. Double-check your requested provider to verify you didn't
simply misspell it.

If you're adding a box from HashiCorp's Atlas, make sure the box is
released.

Name: debian/contrib-jessie64
Address: https://atlas.hashicorp.com/debian/contrib-jessie64
Requested provider: [:docker]
make: *** [dev] Error 1
```

Trying to pop off again again after setting `VAGRANT_DEFAULT_PROVIDER` to `virtualbox`: 

```
$ make dev
make ansible
ansible-galaxy install -r ansible/requirements.yml -p ansible/roles
- geerlingguy.nodejs is already installed, skipping.
vagrant up --provision
Bringing machine 'securethenews' up with 'virtualbox' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* A box must be specified.


make: *** [dev] Error 1
```

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -2,26 +2,27 @@
 # vi: set ft=ruby :
 
 Vagrant.configure("2") do |config|
-
-  config.vm.network "forwarded_port", guest: 8000, host: 8000
-  config.vm.network "forwarded_port", guest: 35729, host: 35729 # livereload
-
   config.vm.define "securethenews" do |securethenews|
+    securethenews.vm.box = "debian/contrib-jessie64"
     securethenews.vm.hostname = "securethenews"
 
-    securethenews.vm.provider "docker" do |d|
+    securethenews.vm.network "forwarded_port", guest: 8000, host: 8000
+    securethenews.vm.network "forwarded_port", guest: 35729, host: 35729 # livereload
+
+    securethenews.vm.provider "docker" do |d, override|
       d.build_dir = 'docker'
       d.has_ssh = true
+
+      # Avoid error caused by defining vm.box for the Docker provider
+      override.vm.box = nil
     end
 
     securethenews.vm.provider "virtualbox" do |vb|
-      config.vm.box = "debian/contrib-jessie64"
       # Building nodejs packages triggers the OOM killer with 512MB of RAM.
       vb.memory = 1024
     end
 
     securethenews.vm.provider "libvirt" do |lv|
-      config.vm.box = "debian/contrib-jessie64"
       lv.memory = 1024
     end
 

